implementations/rust/ockam/ockam_node/src/storage/database/sqlx_from_row_types.rs

```diff
@@ -0,0 +1,80 @@
+use sqlx::database::HasValueRef;
+use sqlx::error::BoxDynError;
+use sqlx::postgres::any::AnyTypeInfoKind;
+use sqlx::{Any, Database, Decode, Type, Value, ValueRef};
+
+/// This type is used to map Option<T> fields for the types deriving `FromRow`
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct Nullable<T>(Option<T>);
+
+impl<T: Clone> Nullable<T> {
+    /// Create a new Nullable value
+    pub fn new(t: Option<T>) -> Self {
+        Nullable(t)
+    }
+
+    /// Return the Option corresponding to this value in the database
+    pub fn to_option(&self) -> Option<T> {
+        self.0.clone()
+    }
+}
+
+impl<'d, T: Decode<'d, Any>> Decode<'d, Any> for Nullable<T> {
+    fn decode(value: <Any as HasValueRef<'d>>::ValueRef) -> Result<Self, BoxDynError> {
+        match value.type_info().kind() {
+            AnyTypeInfoKind::Null => Ok(Nullable(None)),
+            _ => Ok(Nullable(Some(T::decode(value)?))),
+        }
+    }
+}
+
+impl<T: Type<Any>> Type<Any> for Nullable<T> {
+    fn type_info() -> <Any as Database>::TypeInfo {
+        <T as Type<Any>>::type_info()
+    }
+
+    fn compatible(ty: &<Any as Database>::TypeInfo) -> bool {
+        <T as Type<Any>>::compatible(ty) || ty.kind() == AnyTypeInfoKind::Null
+    }
+}
+
+/// This type is used to map boolean fields for the types deriving `FrowRow`.
+/// Postgres provides a proper boolean type but SQLite maps them as integers.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct Boolean(bool);
```

Comment:
Implementing `From` in both ways would have been nice

Reply:
For the same reason as above, it's not so nice because, in the current state, we need to `.clone().into()`.